util/suffix.c

```diff
@@ -31,28 +31,31 @@
  */
 
 #include "suffix.h"
+#include "common.h"
 
 #include <stdlib.h>
 #include <ctype.h>
 #include <errno.h>
 #include <math.h>
+#include <float.h>
+#include <limits.h>
+#include <locale.h>
 
 static struct si_suffix {
 	long double magnitude;
+	unsigned int power;
```

Comment:
Is it really both magnitude and power required? Seems magnitude value can be calcurated from the power without the definition.

Reply:
Yes, properly it works, though I am slightly scared by the math function might result in small errors. Though this pow() should return the correct magnitude.

Hmm, but the name `power` is wrong, should be `exponent` or short `exp`.

Reply:
I think we should keep the magnitude as it avoid pulling in the math library (`libm`). I think the space tradeoff is better.